Building for Windows

### Database name: PostgreSQL

# Issue description

### Describe your problem
Following https://wal-g.readthedocs.io/Windows/, I am trying to cross-compile from debian12. `make pg_build` fails with `undefined: unix.LOCK_SH` -- please see logs/details.

### Please provide steps to reproduce
```bash
git clone https://github.com/wal-g/wal-g.git
cd wal-g/
export GOOS=windows
export GOARCH=amd64
make deps
make pg_build
```

### Please add config and wal-g stdout/stderr logs for debug purpose

<details><summary>If you can, provide logs</summary>
<p>

```bash
(cd main/pg && go build -mod vendor -tags "" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/pg.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/pg.gitRevision=`git rev-parse --short HEAD` -X github.com/wal-g/wal-g/cmd/pg.walgVersion=`git tag -l --points-at HEAD`")
# github.com/wal-g/wal-g/internal/multistorage/stats/cache
../../internal/multistorage/stats/cache/file.go:90:14: undefined: unix.LOCK_SH
../../internal/multistorage/stats/cache/file.go:92:14: undefined: unix.LOCK_EX
../../internal/multistorage/stats/cache/file.go:96:14: undefined: unix.Flock
../../internal/multistorage/stats/cache/file.go:98:18: undefined: unix.EINTR
make: *** [Makefile:44: pg_build] Error 1
```
</p>
</details>

--- a/internal/multistorage/stats/cache/file.go
+++ b/internal/multistorage/stats/cache/file.go
@@ -6,8 +6,6 @@ import (
 	"io"
 	"os"
 	"time"
-
-	"golang.org/x/sys/unix"
 )
 
 type SharedFile struct {
@@ -86,18 +84,3 @@ func (sf *SharedFile) write(content storageStatuses) error {
 	return nil
 }
 
-func lockFile(file *os.File, exclusive bool) (err error) {
-	how := unix.LOCK_SH
-	if exclusive {
-		how = unix.LOCK_EX
-	}
-
-	for {
-		err = unix.Flock(int(file.Fd()), how)
-		// When calling syscalls directly, we need to retry EINTR errors. They mean the call was interrupted by a signal.
-		if err != unix.EINTR {
-			break
-		}
-	}
-	return err
-}

--- a/internal/multistorage/stats/cache/flock_unix.go
+++ b/internal/multistorage/stats/cache/flock_unix.go
@@ -1,0 +1,25 @@
+// +build !windows
+
+package cache
+
+import (
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+func lockFile(file *os.File, exclusive bool) (err error) {
+	how := unix.LOCK_SH
+	if exclusive {
+		how = unix.LOCK_EX
+	}
+
+	for {
+		err = unix.Flock(int(file.Fd()), how)
+		// When calling syscalls directly, we need to retry EINTR errors. They mean the call was interrupted by a signal.
+		if err != unix.EINTR {
+			break
+		}
+	}
+	return err
+}

--- a/internal/multistorage/stats/cache/flock_windows.go
+++ b/internal/multistorage/stats/cache/flock_windows.go
@@ -1,0 +1,44 @@
+// +build windows
+
+package cache
+
+import (
+	"os"
+	"syscall"
+	"unsafe"
+)
+
+var (
+	kernel32, _         = syscall.LoadLibrary("kernel32.dll")
+	procLockFileEx, _   = syscall.GetProcAddress(kernel32, "LockFileEx")
+)
+
+const (
+	winLockfileExclusiveLock   = 0x00000003
+	winLockfileSharedLock      = 0x00000001
+)
+
+func lockFile(file *os.File, exclusive bool) (err error) {
+	how := winLockfileSharedLock
+	if exclusive {
+		how = winLockfileExclusiveLock
+	}
+
+	_, _, errno := syscall.Syscall6(
+		uintptr(procLockFileEx),
+		6,
+		uintptr(syscall.Handle(file.Fd())),
+		uintptr(how),
+		uintptr(0), // reserved
+		uintptr(1), // lock length (low bytes)  |
+		uintptr(0), // lock length (high bytes) |=> entire file
+		uintptr(unsafe.Pointer(&syscall.Overlapped{}))) // offset = 0
+
+	err = nil
+	if errno != 0 {
+		err = errno
+	}
+
+	return err
+}
+

